microcosm_eventsource/immutable_event_factory.py

```diff
@@ -0,0 +1,63 @@
+"""
+Factory for immutable events.
+
+In extension of EventFactory, this class provides a native support of event updating container objects based
+on event types
+
+"""
+from collections import defaultdict
+
+container_event_mutator_registry = defaultdict(list)
+
+
+def register_container_mutator_common(event_type):
+    def decorator(func):
+        container_event_mutator_registry[event_type.__name__].append(func.__name__)
+        return func
+
+    return decorator
+
+
+def register_container_mutator_by_event_type(event_instance_type):
+    def decorator(func):
+        container_event_mutator_registry[event_instance_type.name].append(func.__name__)
+        return func
+
+    return decorator
+
+
+def get_handler_names_for_event_type(event_type):
+    return container_event_mutator_registry[event_type.name]
+
+
+def get_common_handler_names_for_event(event_type):
+    event_type_name = event_type.__class__.__name__.replace('Type', '')
```

Comment:
Is this a valid assumption?

Reply:
You could instead use the `__container__` attribute, and name this `f"{event_type.__class__.__container__.__name__}Event"`

Reply:
Thanks